### ISSUE TITLE
Bidirectional many-to-many

### ISSUE THREAD

**Issue by mindplay-dk** [directus/app#2721]
# Feature Request

I've been trying out the live demo for the following use-case:

I'm building a shop, in which products can have related products.

For example, a "Blue Shirt" is available in small, medium and large - similarly to e.g. Amazon, these will be listed as 3 separate products, and each product page will display links to products related by size.

I created a `size` field as a drop-down, and a `other_sizes` field as a many-to-many:

![image](https://user-images.githubusercontent.com/103348/78118508-66696e00-7407-11ea-8e16-96e95ac78c0d.png)

I created the products and entered the other sizes, but this only works one way:

![image](https://user-images.githubusercontent.com/103348/78118444-4cc82680-7407-11ea-80ca-0dd300eb436d.png)

What is expected for this use-case is a bidirectional relationship: the content manager shouldn't need to duplicate these relationships product by product, and instead the relationships should work in both directions.

## What problem does this feature solve?

Scenarios such as related articles, related products, friendship, and so on - any many-to-many relationship where the direction does not matter.

## How do you think this should be implemented?

This could be implemented as a checkbox or drop-down option in the Many to Many field type.

At the SQL level, there are a few different ways to store and query bidirectional relationships - at least:

1. Store low/high key tuples, e.g. `1, 2` and *not* `2, 1`.

2. Store duplicate tuples, e.g. `1, 2` and `2, 1`, and insert/delete these transactionally.

Where (2) may be slightly easier to query, twice as much storage is required for the records and indices, so I would lean towards (1) personally, e.g.:

```sql
CREATE TABLE other_product_sizes
(
    low_product_id INT NOT NULL,
    high_product_id INT NOT NULL,
    CONSTRAINT one_way CHECK (low_product_id < high_product_id),
    PRIMARY KEY (low_product_id, high_product_id)
);
```

Note that `CHECK` constraints are [new](https://mysqlserverteam.com/mysql-8-0-16-introducing-check-constraint/) to MySQL 8, though (as I recall) the syntax was supported in earlier versions for compatibility, although the constraints weren't actually checked.

Note that my example is self-referencing - it may be similar to directus/directus#2374, not sure, the description/discussion was a bit hard to follow.

I've also seen directus/directus#749, but that appears to be about the UI, whereas (I think) what I'm describing is about the model - if this was a separate field-type (rather than an option for the Many to Many field-type) the UI probably could be identical to that of the Many to Many field.

## Would you be willing to work on this?

I haven't decided to use your product yet, just evaluating - but this feature seemed important enough that it was worth reporting.

It also looks like you're in the middle of rewriting the entire product with proper dependency injection and tests. (Something that will definitely make the product much more attractive to someone like me! although this already looks like a pretty strong product! 🙂)

**Comment by benhaynes** [comment]
To achieve better clarity/visibility, we are now tracking feature requests within the [Feature Request](https://github.com/directus/api/projects/40) project board.

**This issue being closed does not mean it's not being considered.**

**Comment by mattgaspar** [comment]
Maybe this should be filed as a separate bug but it's related to this issue.
When you create a many-to-many field and on the relation section you select the same collection and choose Auto-Generate, it gets populated with duplicate field id's and the api throws an error when you try to save.

![image](https://user-images.githubusercontent.com/811116/78616210-75498980-7828-11ea-84ab-2c118e574b2d.png)

If one of the id's is changed, that would basically be creating what @mindplay-dk described as a "directed many-to-many relationship" because the relationship can only be seen from one side of the relationship.  But this seems to not work, it broke the collection on the demo site! (I had changed the second staff_id to staff_id2) Unless the field id's get reset on the backend when auto-generate is selected.

![image](https://user-images.githubusercontent.com/811116/78617040-d6725c80-782a-11ea-9fcc-414ece01e858.png)
